### PR TITLE
addinf features for number lookup

### PR DESCRIPTION
**Pull request by rajjy1**

### CODE REVIEW

File: src/Telnyx.Example/Program.cs [hunk 1]
```diff
@@ -96,6 +96,12 @@ static async Task Main(string[] args)
             callControlApplicationsExample.Get();
             callControlApplicationsExample.Delete();
             callControlApplicationsExample.Update();
+
+
+            Console.WriteLine("NumberLookupExample...");
+            NumberLookupExample numberLookupExample = new NumberLookupExample();
+            numberLookupExample.List();
+            numberLookupExample.ListAsync();
```
Comment (thedonmon) [inline src/Telnyx.Example/Program.cs:104]: add await to line 104

File: src/Telnyx.Example/NumberLookupExample.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }
+            catch (TelnyxException ex)
+            {
+                Console.WriteLine("exception");
+                Console.WriteLine(JsonConvert.SerializeObject(ex));
+            }
+            return list;
+        }
+
+        public Task<TelnyxList<NumberLookupRecord>> ListAsync()
```
Comment (thedonmon) [inline src/Telnyx.Example/NumberLookupExample.cs:38]: missing async modifier

File: src/Telnyx.Example/Program.cs (hunk 1 above)
Comment (rajjy1) [reply src/Telnyx.Example/Program.cs:104]: Added.

File: src/Telnyx.Example/NumberLookupExample.cs (hunk 2 above)
Comment (rajjy1) [reply src/Telnyx.Example/NumberLookupExample.cs:38]: added.

**Comment by rajjy1** [comment]
> looks good a couple of small additions needed to complete this but also missing tests for new feature.

Added unit tests as well.

**Pull request merged**